src/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php

```diff
@@ -45,6 +45,26 @@ public function process(File $phpcsFile, $stackPtr)
     {
         $tokens = $phpcsFile->getTokens();
 
+        // Bail when handling the most common cases of a short list inside a foreach (see
+        // https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/527).
+        if ($tokens[$stackPtr]['code'] === T_OPEN_SHORT_ARRAY
+            && isset($tokens[$stackPtr]['nested_parenthesis']) === true
+        ) {
+            $lastParenthesisCloser = end($tokens[$stackPtr]['nested_parenthesis']);
+            $lastParenthesisOpener = key($tokens[$stackPtr]['nested_parenthesis']);
+            reset($tokens[$stackPtr]['nested_parenthesis']);
```

Comment:
Except that `foreach` acts on a _copy_ of the array and doesn't look at the internal array pointer at all, so sorry, that argument falls short.

Reply:
Now I can see my mistake. I tested what I said above last week using PsySH:

![Screenshot from 2024-12-02 15-47-29](https://github.com/user-attachments/assets/4072cdb6-b8b4-4936-85aa-103d0fee9439)

I accidentally pressed "Enter" between the `foreach` and the opening brace. What I didn't realize was that because of this, the `echo` statement was executed outside of the `foreach`. When I saw that only `5` was printed, I got to the incorrect conclusion that I mentioned above. Or at least, this is my current understanding of what happened.

Still, don't you think that it is a good practice to leave the array pointer pointing to the initial value? I believe other uses of the array will not use a copy and will use the internal array pointer. I can see that in other places in the PHPCS codebase where `end()` and `key()` are used, `reset()` is not called, so it is probably fine.

Reply:
IIRC the internal array pointer is only used by the old-school `end()`, `key()`, `current()`, `prev()`, `next()`, `reset()` and `each()` functions.
Those should rarely be used in modern code.

Either way, to be honest, this whole discussion is academic as per commit c149b391d7a5add831886ad07972a96bff031556, `end()` is now acting on a _copy_ of the sub-array.

Though I noticed that commit c149b391d7a5add831886ad07972a96bff031556
 is incomplete and contains a bug now as `key()` is not using the sub-array, while it should...

In other words, the `$lastParenthesisOpener = key($tokens[$stackPtr]['nested_parenthesis']);` line needs to be fixed, then the `reset()` is 100% safe to remove.

Reply:
Thanks for the additional details, and I'm sorry I missed that the `$lastParenthesisOpener` also needed to be updated. This PR should be ready for a final review.